src/ui/analytics/chart/linearChart.ui.tsx

```diff
@@ -0,0 +1,108 @@
+import Image from 'next/image';
+import { useEffect, useRef } from 'react';
+import { getMetaverseColor } from '../../../utils/analyticsChart';
+import { LightweightLinearChart } from '../../../lib/lightweightLinearChart.util';
+import { Currencies, Metaverses } from '../../../enums/common';
+import { AnalyticChartData } from '../../../interfaces/charts';
+import { convertCurrency } from '../../../utils/currencyConverter';
+import { useSelector } from 'react-redux';
+import { RootState } from '../../../state/store';
+
+// Interface for props used in the LinearChartUI component.
+interface LinearChartUIProps {
+  chartApiData: AnalyticChartData; // Data for the linear chart
+  chartOptions: {
+    cols: 'col-span-1' | 'col-span-2'; // Column span options for chart layout
+    visibleDays: number; // Number of visible days for the chart
+    visibleMetaverses: Metaverses[]; // Array of visible metaverses for the chart
+    currencySelected: Currencies; // Selected currency for the chart
+  };
+}
+
+//* Main component for the linear chart.
```

Comment:
this comment doesn't add util info to component (maybe you want mark the main component, it is not necessary becouse you should have the components on differents files)